### PR TITLE
Don't cache subjects queue

### LINKED ISSUE #355: Reload pool diminished
This was originally over in issue #346, but is potentially a separate issue. Whereas some people such as @charlesdorsett are seeing some pages they have already transcribed, Mutabilitie is trying to refresh pages and load new ones without transcribing them. It's a soft skip in a way meaning the page will come back later. A number of volunteers have been doing this over the last two years and it used to work, but isn't working now. If subjects are truly served randomly, a hard refresh should get something out of your queue. The pool M is seeing is stuck at around 2-3 dozen images. I think this is possibly linked to issue #346, from what I understand of the way subjects are supposed to be served. 

M writes: 

"Basically, whenever I'm being shown a document that I don't fancy at the moment, I reload https://www.shakespearesworld.org/#!/transcribe/2778, which will pull a new document from the pool (I prefer that to just clicking "submit" right away, because a lot of the time I'll be happy to transcribe that page when it comes up again at a later point). Until a few months ago, this worked fine, because there were lots of documents in the pool and I would easily find a new one that I preferred after reloading two or three times. Now, though, the entire pool seems to consist of about 2-3 dozen images, which I have already seen lots of times. This means that when I reload https://www.shakespearesworld.org/#!/transcribe/2778, I will keep being shown the same 2-3 dozen images, until I flush them out of the pool, as it were by clicking "submit" (with or without transcribing a token like). I'm positive that this wasn't an issue before."

Sample subject images at issue include 53901, 53894, and 53711. These are just a few examples.

### PR DESCRIPTION
**Pull request by simoneduca**
Fix #355

### CODE REVIEW

**Comment by simoneduca** [comment]
@camallen mind checking the behaviour here? https://preview.zooniverse.org/shakespearesworld

**Comment by camallen** [comment]
@simoneduca that link doesn't use local storage for subjects, however it doesn't set the current subject and it persists the annotations across subject reloads thus might incorrectly submit annotations for the wrong subject. 

If i start annotating a subject should it remember this and ask me before reloading / reload that subject that i've been working on?

**Comment by simoneduca** [comment]
Ah yes, my bad. I had that check/warning system in place already, which is made useless by my latest changes. I'll store the current subject and clear once the work is submitted or cancelled.

**Comment by simoneduca** [comment]
@camallen and I have figured out what was wrong with the subjects queue. I've just published the latest changes to https://preview.zooniverse.org/shakespearesworld/#!/
@vvh please test if the app behaves as expected now. Please note that on first loading the page you may see the "unsaved work" warning. We're trying to fix that too.

**Comment by VVH** [comment]
This seems to be working fine for me, but it would be best to ask those who reported the bug on Talk, including mutabilitie, I think?

**Comment by camallen** [comment]
I've messaged mutabilitie asking her to check.

**Comment by simoneduca** [comment]
Thanks Cam.

File: app/modules/transcribe/subjects.factory.js [hunk 1]
```diff
@@ -161,7 +153,7 @@ function SubjectsFactory($q, AnnotationsFactory, localStorageService, zooAPI, zo
     }
 
     function _setCurrent() {
-        _data.current = _queue.shift();
+        _data.current = _queue[Math.floor(Math.random() * _queue.length)];
```
Comment (eatyourgreens) [inline app/modules/transcribe/subjects.factory.js]: `Array.shift()` removes the first element from an array, but the new code here doesn't change the `_queue` array. Is that intentional? It looks like you will always serve the same small group of subjects to the volunteers with this code.

Comment (simoneduca) [reply app/modules/transcribe/subjects.factory.js]: Ah I just missed that, thanks! I'll push a fix

Comment (simoneduca) [reply app/modules/transcribe/subjects.factory.js]: @eatyourgreens please check last commit

File: app/modules/transcribe/subjects.factory.js [hunk 2]
```diff
@@ -161,7 +153,8 @@ function SubjectsFactory($q, AnnotationsFactory, localStorageService, zooAPI, zo
     }
 
     function _setCurrent() {
-        _data.current = _queue.shift();
+        var randonIndex = Math.floor(Math.random() * _queue.length);
```
Comment (eatyourgreens) [inline app/modules/transcribe/subjects.factory.js]: typo in random.

Comment (simoneduca) [reply app/modules/transcribe/subjects.factory.js]: Thanks

**Pull request merged**